lms/static/scripts/frontend_apps/components/GroupConfigSelector.js

```diff
@@ -33,28 +33,35 @@ import ErrorModal from './ErrorModal';
  * @prop {GroupSet[]|null} groupSets
  * @prop {(id: string|null) => void} onInput
  * @prop {string|null} selectedGroupSetId
+ * @prop {string} productFamily - Which product (canvas, blackboard...) we are running in.
  *
  * @param {GroupSelectProps} props
  */
-function GroupSelect({ busy, groupSets, onInput, selectedGroupSetId }) {
+function GroupSelect({
+  busy,
+  groupSets,
+  onInput,
+  selectedGroupSetId,
+  productFamily,
+}) {
   const selectId = useUniqueId('GroupSetSelector__select');
 
   return (
     <>
-      <label htmlFor={selectId}>Group set: </label>
-      <select
+      <GroupSelectHeader htmlFor={selectId} productFamily={productFamily} /> <select
```

Comment:
Do you foresee anything other than the text changing for the components other than the error dialog. If not, perhaps it would make sense to build a simple message ID / message text lookup, with a mechanism to support adjusting the language depending on the product.

```js
const messages = {
  selectLabel: 'Group set: ', // Default
  selectLabel_desire2learn: 'Group categories: ' // Customized for d2l
};
const message = id => messages[id + "_" + productFamily] ?? messages[id];
...
<label htmlFor={selectId}>{message("selectLabel")}</label>
```